### PR TITLE
ttl: retry delete when write conflict happens in txn

### LINKED ISSUE #40502: TTL delete sql should retry when write conflict happens

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #40502

### What is changed and how it works?

In previous implement, the delete operation in TTL will retry when error occurs. But it will not retry when commit failed. In some cases, there will be a write conflict when committing the delete transaction while another user session are updating the record, and we need retry the deletion at this time.

The PR also retries the delete sql when commit failed. 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ttl/ttlworker/session.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ type ttlTableSession struct {
 	expire time.Time
 }
 
-func (s *ttlTableSession) ExecuteSQLWithCheck(ctx context.Context, sql string) (rows []chunk.Row, shouldRetry bool, err error) {
+func (s *ttlTableSession) ExecuteSQLWithCheck(ctx context.Context, sql string) (rows []chunk.Row, _ bool, err error) {
```
Comment (mjonss) [inline ttl/ttlworker/session.go:155]: Please keep the output variable or remove all output variables and declare `rows` before `s.RunInTxn`

**Review by mjonss — APPROVED** [review]
LGTM with one minor suggestion.

File: ttl/ttlworker/session.go (hunk 1 above)
Comment (lcwangchao) [reply ttl/ttlworker/session.go]: done

**Review by YangKeao — APPROVED** [review]
LGTM

**Pull request merged**